providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithClientConfigurer.java

```diff
@@ -0,0 +1,169 @@
+package dev.openfeature.contrib.providers.flagsmith;
+
+import com.flagsmith.FlagsmithClient;
+import com.flagsmith.config.FlagsmithCacheConfig;
+import com.flagsmith.config.Retry;
+import dev.openfeature.contrib.providers.flagsmith.exceptions.InvalidCacheOptionsException;
+import dev.openfeature.contrib.providers.flagsmith.exceptions.InvalidOptionsException;
+
+/**
+ * FlagsmithClientConfigurer helps set up and validate the options for the FlagsmithClient
+ * used by the FlagsmithProvider class.
+ */
+public class FlagsmithClientConfigurer {
+
+    /**
+     * initializeProvider is initializing the different class element used by the provider.
+     *
+     * @param options the options used to create the provider
+     */
+    static FlagsmithClient initializeProvider(FlagsmithProviderOptions options) {
+
+        validateOptions(options);
+
+        FlagsmithClient.Builder flagsmithBuilder = FlagsmithClient
+            .newBuilder();
+        // Set main configuration settings
+        if (options.getApiKey() != null) {
```

Comment:
This check seems a little redundent with the `validateOptions` above

Reply:
Agreed, it's been removed